conda/recipes/cudf/meta.yaml

```diff
@@ -30,7 +30,7 @@ requirements:
   run:
     - python
     - pandas>=0.24.2,<0.25
-    - cupy >=6.5.0,<7
+    - cupy >=6.5.0,<=7.0.0
```

Comment:
I think the only thing missing in the PR is to update the conda dev environments to coincide with the version here

Reply:
Should we start requiring 6.6.0 so we get the fix for __cuda_array_interface__?
Since it's been out for a while, I'm not sure there is an advantage to allowing 6.5.0 support, especially when it adds confusion for users.

Reply:
if we remove the patch I think we need to require 6.6.0+.

Reply:
Yeah no objections to requiring 6.6.0.

I think I would make this more flexible actually and allow any version less than 8.

Reply:
Have updated the environments as well. Also made the lower bound CuPy 6.6.0. Finally relaxed the upper bound to anything less that CuPy 8.0.0.